SplunkRumWorkspace/SplunkRum/SplunkRum/Session.swift

```diff
@@ -54,14 +56,25 @@ func getRumSessionId() -> String {
     }
     if Date() > sessionIdExpiration {
         sessionIdExpiration = Date().addingTimeInterval(TimeInterval(MAX_SESSION_AGE_SECONDS))
+        oldRumSessionId = rumSessionId
         rumSessionId = generateNewSessionId()
+        isSessionIdChanged = true
```

Comment:
Can we just call `createSessionIdChangeSpan` here and get rid of the boolean?

Reply:
Let me explain what happens if we call` createSessionIdChangeSpan` in `getRumSessionId`.
In `getRumSessionId` when new session id is generated , we use `sessionIdLock.lock()`, if we call `createSessionIdChangeSpan` from `getRumSessionId` then agin it call `getRumSessionId`  to assign session id to newly created span , but `sessionIdLock` is already lock so dead lock situation will occur because of that app get crashed , so that we used `isSessionIdChanged` flag and after unlock the sessionIdLock , we Called `createSessionIdChangeSpan`